docs/model_developers/modules.rst

```diff
@@ -347,10 +347,26 @@ Add the following content to the file:
   login <username>
   password <password>
 
-For more information see the doc about`pip authentication <https://pip.pypa.io/en/stable/topics/authentication/>`_.
+For more information see the doc about `pip authentication <https://pip.pypa.io/en/stable/topics/authentication/>`_.
 
 You will also need to specify the url of the repository in the ``project.yml`` file of your project (See: :ref:`specify_location_pip`).
 
 By following the previous steps, the Inmanta server will be able to install modules from a private Python package repository.
 
 
+Inter-module dependencies
+#########################
+
+The plugins code of a module A can have a dependency on the plugins code of another V2 module B. When doing this,
+care should be taken that the Python module(s) you depend on, do not define any resources or providers. Otherwise the
+python environment of the agent can get corrupt in the following way:
+
+1. Module A-1.0 depends on X.py of module B-1.0
+2. A-1.0 and B-1.0 are exported: The exporter exports the X.py file to the server and the agent puts the X.py file in
+   its code directory.
+3. In a later version of module B (B-2.0) the X.py file doesn't have any resources anymore.
```

Comment:
I think this one should be split up:
- the resources being present or not are a property of the model, which will in most (if not all) cases be on the side of either the project or module `A`.
- Then, sometime after that a new version of module `B` is released and included in the project.

Reply:
I don't understand this. Could you elaborate?

Reply:
I see that usually there is also a change required on module mod-a, but I still don't see clearly which changes you want to see applied. Could you make a concrete proposal?

Reply:
1. <current>. The configuration model (in the project or one of the modules) constructs resources from both modules A and B.
2. <unchanged>
3. The configuration model is changed to only construct resources from module A.
4. A-1.0 is exported again, B-1.0 is no longer exported because it doesn't have any resources. The X.py file still exists in the agent code directory.
5. A new version of module B (B-2.0) is released and included in the project. The project is re-exported: A-1.0 is exported again, B-2.0 is not (again because it doesn't have any resources). The old X.py file still exists in the agent code directory. It is loaded by the agent instead of the one from B-2.0.

It still cuts some corners but I think it is an OK compromise.